libwallet/internal/politeia/politeia.go

```diff
@@ -90,28 +91,45 @@ func (p *Politeia) saveOrOverwiteProposal(proposal *Proposal) error {
 }
 
 // GetProposalsRaw fetches and returns a proposals from the db
-func (p *Politeia) GetProposalsRaw(category int32, offset, limit int32, newestFirst bool) ([]Proposal, error) {
-	return p.getProposalsRaw(category, offset, limit, newestFirst, false)
+func (p *Politeia) GetProposalsRaw(category int32, offset, limit int32, newestFirst bool, key string) ([]Proposal, error) {
+	return p.getProposalsRaw(category, offset, limit, newestFirst, false, key)
 }
 
-func (p *Politeia) getProposalsRaw(category int32, offset, limit int32, newestFirst bool, skipAbandoned bool) ([]Proposal, error) {
+func (p *Politeia) getProposalsRaw(category int32, offset, limit int32, newestFirst bool, skipAbandoned bool, searchKey string) ([]Proposal, error) {
 	var query storm.Query
+	searchKey = strings.TrimSpace(strings.ToLower(searchKey))
 	switch category {
 	case ProposalCategoryAll:
 
 		if skipAbandoned {
-			query = p.db.Select(
-				q.Not(q.Eq("Category", ProposalCategoryAbandoned)),
-			)
+			if searchKey != "" {
+				query = p.db.Select(
+					q.Not(q.Eq("Category", ProposalCategoryAbandoned)),
+					q.Re("LowerName", "^"+searchKey),
+				)
+			} else {
+				query = p.db.Select(
+					q.Not(q.Eq("Category", ProposalCategoryAbandoned)),
+				)
+			}
 		} else {
-			query = p.db.Select(
-				q.True(),
-			)
+			if searchKey != "" {
+				query = p.db.Select(
+					q.True(),
+					q.Re("LowerName", "^"+searchKey),
```

Comment:
First. Prefixing the search phrase with "^" means it will only match proposals whose name begin with the search phrase, I think. It should match proposals that have the search expression in any part of the proposal name, not just the beginning.

Second, you can match the `Name` field in a case-insensitive manner, you won't need a `LowerName` field. Prefix the query with "(?i)" and that should achieve a case-insensitive regex matching.